opend/src/opend.d

```diff
@@ -0,0 +1,160 @@
+module opend;
+
+import std.algorithm;
+import std.getopt;
+import std.file;
+import std.string : endsWith;
+import std.path;
+import std.process : execute;
+import std.stdio : writeln;
+
+string compiler = `dmd`;
+string type = "executable";
+string buildType = "debug";
+
+/// Retrieves all files as full path
+string[] getFiles(string dir)
+{
+    string[] ret;
+    foreach(ref DirEntry d; dirEntries(dir, SpanMode.breadth))
+    {
+        // Only care about files with ".d" extensions
+        if(d.isDir || extension(d.name) != ".d")
+            continue;
+
+        // Auto-escape pathes!
+        // Actually, I checked (on Windows) and paths having spaces in them work fine too.
+        ret ~= /*"\"" ~ */absolutePath(d.name) /*~ "\""*/;
+    }
+    return ret;
+}
+
+string getExeFileName()
+{
+    immutable dname = baseName(getcwd());
+    writeln(dname);
+    version(Windows)
+    {
+        immutable extension = ".exe";
+    }
+    else
+    {
+        immutable extension = "";
+    }
+
+    return dname ~ extension;
+}
+
+string getLibFileName()
+{
+    immutable dname = baseName(getcwd());
+    version(Windows)
+    {
+        immutable extension = ".lib";
+    }
+    else
+    {
+        immutable extension = "";
```

Comment:
Why not use the `.o` extension here? Otherwise the static library would have the same name as the binary if it was built.